maint/yutils.py

```diff
@@ -25,6 +25,7 @@ def copyright(outfile):
 ##### generate an array of datatype arrays
 ########################################################################################
 def generate_darrays(derived_types, darraylist, maxlevels):
+    darraylist.append([])
```

Comment:
Why do we need this empty entry?

Reply:
Earlier, all levels were being generated dynamically.  So we don't have a use case for all levels being "NULL" (basically the builtin case), because that was handled separately as a direct memcpy.

Now we have separated out two of the nesting levels to be generated statically and generate the remaining levels dynamically.  So the dynamically generated levels can have both "NULL" (meaning that only the top two levels are considered for the pack/unpack kernels).